src/block-ciphers.rst

```diff
@@ -12,109 +12,103 @@ Block ciphers
+fixed length. The encryption function :math:`E` transforms
+plaintext blocks :math:`P` into ciphertext blocks :math:`C` by using a
 secret key :math:`k`:
 
 .. math::
 
    C = E(k, P)
 
-The plaintext and ciphertext blocks are sequences of bits. They are
-always the same size as one another, and that size is fixed by the block
-cipher: it's called the block cipher's *block size*. The set of all
-possible keys is called the :term:`keyspace`.
+Plaintext and ciphertext blocks are sequences of bits and always match in size. The 
+block cipher's *block size* is a fixed size. :term:`Keyspace` is the set of all possible keys.
 
-Once we've encrypted plaintext blocks into ciphertext blocks, they later
-have to be decrypted again to recover the original plaintext block. This
-is done using a decryption function :math:`D`, which takes the
+Once we encrypt plaintext blocks into ciphertext blocks, they are later
+decrypted to recover original plaintext block. The original plaintext block
+:math:`P` is produced using a decryption function :math:`D`. It takes the
 ciphertext block :math:`C` and the key :math:`k` (the same one used to
-encrypt the block) as inputs, and produces the original plaintext block
-:math:`P`.
+encrypt the block) as inputs.
 
 .. math::
 
    P = D(k, C)
 
-Or, in blocks:
+Or, visually represented in blocks:
 
 .. figure:: Illustrations/BlockCipher/BlockCipher.svg
    :align: center
 
-Block ciphers are an example of a :term:`symmetric-key encryption` scheme, also
-known as a :term:`secret-key encryption` scheme. This means that the same secret
-key is used for both encryption and decryption. We will contrast this
+A block cipher is an example of a :term:`symmetric-key encryption` scheme, also
+known as a :term:`secret-key encryption` scheme. The same secret
+key is used for both encryption and decryption. Later in the book, we contrast this
 with :term:`public-key encryption` algorithms, which have a distinct key for
-encryption and decryption, later in the book.
-
-A block cipher is a *keyed permutation*. It's a *permutation*, because
-the block cipher maps every possible block to some other block. It's
-also a *keyed* permutation, because the key determines exactly which
-blocks map to which. It's important that it's a permutation, because the
-recipient needs to be able to map blocks back to the original blocks as
-well, which you can only do if it's one-to-one.
-
-We'll illustrate this by looking at a block cipher with an impractical,
-tiny 4-bit block size, so :math:`2^4 = 16` possible blocks. Since each
-of those blocks maps to a hexadecimal digit, we'll represent the blocks
+encryption and decryption.
+
+A block cipher is a *keyed permutation*. First, the key determines exactly which
+blocks are mapped. Second, the permutation aspect allows mapping of all possible blocks 
+to other blocks. Permutation is important because the recipient must map blocks back to the 
+original blocks by analyzing one-to-one.
```

Comment:
Not entirely sure about this paragraph.

>  First, the key determines exactly which blocks are mapped.

That's true, but I'm worried that it implies that not all blocks are mapped under all keys, but that's not true. It changes _which blocks map to which blocks_, but all blocks are always mapped to something.

> Second, the permutation aspect allows mapping of all possible blocks to other blocks. Permutation is important because the recipient must map blocks back to the original blocks by analyzing one-to-one.

I don't really understand "by analyzing one-to-one.", maybe without it'd be fine? Or "must be able to map blocks back..."?